ai21_tokenizer/base_jurassic_tokenizer.py

```diff
@@ -0,0 +1,255 @@
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional, Dict, Any, BinaryIO, Tuple, Union
+
+import sentencepiece as spm
+
+from abc import ABC
+from ai21_tokenizer.utils import is_number, PathLike, load_json
+
+_MODEL_EXTENSION = ".model"
+_MODEL_CONFIG_FILENAME = "config.json"
+
+
+@dataclass
+class SpaceSymbol:
+    tok_id: int
+    count: int
+
+
+class BaseJurassicTokenizer(ABC):
+    _sp: spm.SentencePieceProcessor = None
```

Comment:
Type is wrong. If something could be `None` then its type is `Optional[<object_type>]`